mini-projects/00-jq-rs/src/processor/operations.rs

```diff
@@ -0,0 +1,214 @@
+use serde_json::Value;
+//Enum representing the different outputs we could get from an operation
+#[derive(PartialEq, Debug)]
+pub enum Result {
+    String(String),
+    Value(Value),
+    Lenght(usize),
+}
+pub trait Operate {
+    fn operate(&mut self) -> Result;
+}
+pub struct Add {
+    input: Value,
+}
+impl Add {
+    pub fn new(input: Value) -> Self {
+        let output = Add { input: input };
+        output
+    }
+}
+//Implements adding functionality for all potential value types
+impl Operate for Add {
+    fn operate(&mut self) -> Result {
+        let mut output: String = String::new();
+        for val in self.input.as_array().unwrap() {
+            match val {
+                Value::Number(num) => output.push_str(&num.to_string()),
+                Value::String(string) => output.push_str(string),
+                Value::Array(arr) => {
+                    for val in arr {
+                        output.push_str(&val.as_str().unwrap());
+                    }
+                }
+                Value::Bool(bool) => output.push_str(&bool.to_string()),
+                Value::Object(obj) => {
+                    for (key, value) in obj {
+                        output.push_str(&key);
+                        output.push_str(" ");
+                        output.push_str(&value.to_string());
+                    }
+                }
+                Value::Null => output.push_str("null"),
+            };
+        }
+        Result::String(output)
+    }
+}
+pub struct Delete {
+    input: Value,
+    needle: String,
+}
+impl Delete {
+    pub fn new(input: Value, needle: String) -> Self {
+        let output = Delete {
+            input: input,
+            needle: needle,
+        };
+        output
+    }
+}
+impl Operate for Delete {
```

Comment:
I like this - Operate seems like a somewhat poor naming choice - I know what your going for, but operate is a pretty ambiguous term